### PR TITLE
feat: Add compile-time storage cost lookup table for O(1) gas calculation

### PR DESCRIPTION
**Pull request by roninjin10**
## Summary
- Implements compile-time storage cost lookup table for SSTORE operations
- Provides O(1) gas calculation to eliminate runtime conditionals
- Adds comprehensive support for all Ethereum hardforks
- Includes optional size optimization for embedded targets

## Key Components
- **StorageStatus enum**: Defines storage state transitions (Unchanged, Added, Deleted, Modified)
- **StorageCost struct**: Contains gas cost and refund values
- **STORAGE_COST_TABLE**: Pre-computed lookup table indexed by hardfork and status
- **getStorageCost()**: O(1) lookup function
- **Build option**: `-Doptimize_size=true` for runtime calculation mode

## Performance Benefits
- Direct array access instead of multiple runtime conditionals
- Improved CPU pipeline utilization
- Cache-friendly sequential memory access
- ~2KB binary size increase (optional)

## Hardfork Support
Correctly implements gas costs for all major Ethereum hardforks:
- Pre-Constantinople: Basic costs with 15000 gas refunds
- Istanbul (EIP-2200): Rebalanced SSTORE costs
- Berlin (EIP-2929): Cold/warm access patterns  
- London (EIP-3529): Reduced refund amounts
- All subsequent forks (Merge, Shanghai, Cancun)

## Test Coverage
- Comprehensive unit tests for all status types and hardforks
- Integration tests with existing SSTORE implementation
- Edge case validation (zero values, same values, etc.)
- Both table-based and runtime calculation modes tested

## Backward Compatibility
- Fully backward compatible with existing code
- No changes to public APIs
- All existing tests continue to pass

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -136,10 +136,14 @@ pub fn build(b: *std.Build) void {
 
     // Custom build option to disable precompiles
     const no_precompiles = b.option(bool, "no_precompiles", "Disable all EVM precompiles for minimal build") orelse false;
+    
+    // Custom build option to optimize for size
+    const optimize_size = b.option(bool, "optimize_size", "Optimize for binary size by using runtime calculations instead of lookup tables") orelse false;
```
Comment (roninjin10) [inline build.zig:141]: @claude we need to use the builtin optimizeSize not add our own
